core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -386,7 +387,7 @@ public final void runInContext() {
             }
           } catch (Throwable t) {
             stream.cancel(Status.CANCELLED.withCause(t).withDescription("Failed to read message."));
-            return;
+            Throwables.propagateIfPossible(t);
```

Comment:
Exception propagation is little different here, since we don't really have a full callstack.  The exception should get propagated back to the ClientCall.Listener provided at the start of the call, with the exception in the params of onClose().  If that is not the case, it would be a bug and that should be fixed. 

 The executor doing the work here doesn't care if the runnable threw, because it really can't do anything about it.  It logs the message as a last ditch effort, but that's really only to catch places where the throwable wasn't properly propagated down the stream.